libnavigation-core/src/main/java/com/mapbox/navigation/core/reroute/MapboxRerouteController.kt

```diff
@@ -57,7 +57,7 @@ internal class MapboxRerouteController @VisibleForTesting constructor(
         MapboxRerouteOptionsAdapter(evDynamicDataHolder)
     )
 
-    private var rerouteOptionsAdapter: RerouteOptionsAdapter? = defaultRerouteOptionsAdapter
+    private var externalRerouteOptionsAdapter: RerouteOptionsAdapter? = null
```

Comment:
I don't like it that now we have:
1) 2 chained `RerouteOptionsAdapter`s here;
2) an arbitrary number of chained `RerouteOptionsAdapter`s in `MapboxRerouteOptionsAdapter`.

I think that's confusing and we should keep all the different adapters in one place (`MapboxRerouteOptionsAdapter`). Do you mind if I refactor it a bit?

Reply:
Sounds good :+1: